File: CPP/Program/Vehicle.cpp
// Menggunakan pragma once agar header file hanya di-include satu kali
#pragma once

// Memasukan library yang digunakan
#include <iostream>
#include <string>

// Deklarasi namespace
using namespace std;

// Kelas Vehicle digunakan untuk mempresentasikan data Kendaraan
// Kelas Vehicle merupakan kelas induk dari kelas Car dan kelas Motorcycle
class Vehicle {
    private:
        // Atribut yang digunakan pada kelas Vehicle terdiri dari nama kendaraan, plat nomor, merk, tahun produksi, dan warna. 
        string nama_kendaraan;
        string plat_nomor;
        string merk;
        string tahun_produksi;
        string warna;

    public:
        // Constructor
        Vehicle() {

        }

        // Constructor : Overloading
        Vehicle(string nama_kendaraan) {
            this->nama_kendaraan = nama_kendaraan;
            this->plat_nomor = "";
            this->merk = "";
            this->tahun_produksi = "";
            this->warna = "";
        }

        // Constructor : Overloading
        Vehicle(string nama_kendaraan, string plat_nomor) {
            this->nama_kendaraan = nama_kendaraan;
            this->plat_nomor = plat_nomor;
            this->merk = "";
            this->tahun_produksi = "";
            this->warna = "";
        }

        // Constructor : Overloading
        Vehicle(string nama_kendaraan, string plat_nomor, string merk) {
            this->nama_kendaraan = nama_kendaraan;
            this->plat_nomor = plat_nomor;
            this->merk = merk;
            this->tahun_produksi = "";
            this->warna = "";
        }

        // Constructor : Overloading
        Vehicle(string nama_kendaraan, string plat_nomor, string merk, string tahun_produksi) {
            this->nama_kendaraan = nama_kendaraan;
            this->plat_nomor = plat_nomor;
            this->merk = merk;
            this->tahun_produksi = tahun_produksi;
            this->warna = "";
        }

        // Constructor : Overloading
        Vehicle(string nama_kendaraan, string plat_nomor, string merk, string tahun_produksi, string warna) {
            this->nama_kendaraan = nama_kendaraan;
            this->plat_nomor = plat_nomor;
            this->merk = merk;
            this->tahun_produksi = tahun_produksi;
            this->warna = warna;
        }

        // Method setter untuk set nilai setiap atribut pada kelas Vehicle
        void setNamaKendaraan(string nama_kendaraan) {
            this->nama_kendaraan = nama_kendaraan;
        }

        void setPlatNomor(string plat_nomor) {
            this->plat_nomor = plat_nomor;
        }

        void setMerk(string merk) {
            this->merk = merk;
        }

        void setTahunProduksi(string tahun_produksi) {
            this->tahun_produksi = tahun_produksi;
        }

        void setWarna(string warna) {
            this->warna = warna;
        }

        // Method getter untuk mendapatkan nilai setiap atribut pada kelas Vehicle
        string getNamaKendaraan() {
            return nama_kendaraan;
        }
        
        string getPlatNomor() {
            return plat_nomor;
        }

        string getMerk() {
            return merk;
        }

        string getTahunProduksi() {
            return tahun_produksi;
        }

        string getWarna() {
            return warna;
        }

        // Destructor
        ~Vehicle() {

        }
};

File: Python/Program/Vehicle.py
# Kelas Vehicle digunakan untuk mempresentasikan data Kendaraan
# Kelas Vehicle merupakan kelas induk dari kelas Car dan kelas Motorcycle
class Vehicle:
    # Atribut yang digunakan pada kelas Vehicle terdiri dari nama kendaraan, plat nomor, merk, tahun produksi, dan warna. 
    # Constructor
    def __init__(self, nama_kendaraan="", plat_nomor="", merk="", tahun_produksi="", warna=""):
        self.nama_kendaraan = nama_kendaraan
        self.plat_nomor = plat_nomor
        self.merk = merk
        self.tahun_produksi = tahun_produksi
        self.warna = warna
        
    # Method setter untuk set nilai setiap atribut pada kelas Vehicle
    def setNamaKendaraan(self, nama_kendaraan):
        self.nama_kendaraan = nama_kendaraan

    def setPlatNomor(self, plat_nomor):
        self.plat_nomor = plat_nomor

    def setMerk(self, merk):
        self.merk = merk

    def setTahunProduksi(self, tahun_produksi):
        self.tahun_produksi = tahun_produksi

    def setWarna(self, warna):
        self.warna = warna

    # Method getter untuk mendapatkan nilai setiap atribut pada kelas Vehicle
    def getNamaKendaraan(self):
        return self.nama_kendaraan

    def getPlatNomor(self):
        return self.plat_nomor

    def getMerk(self):
        return self.merk

    def getTahunProduksi(self):
        return self.tahun_produksi

    def getWarna(self):
        return self.warna
        
    

    